src/Symfony/Component/Workflow/Workflow.php

```diff
@@ -71,7 +73,11 @@ public function getMarking(object $subject)
             // update the subject with the new marking
             $this->markingStore->setMarking($subject, $marking);
 
-            $this->entered($subject, null, $marking);
+            if (empty($context)) {
```

Comment:
```suggestion
            if (!$context) {
```

But I'm not sure about this if. This is very opinionated. It could be added everytime, what do you think?

Reply:
I like the word `empty` as it is more explicit for an array. $context can't be a boolean, nor an object.
But I'll change if you want to.

Reply:
Yes please